File: scripts/fix_module_path.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');

const OLD = 'github.com/TeaOSLab/EdgeAdmin';
const NEW = 'github.com/hujiali30001/freecdn-admin';
const ROOT = path.resolve(__dirname, '..');

let changedFiles = 0, totalReplacements = 0;

function walk(dir) {
  for (const name of fs.readdirSync(dir)) {
    if (name === '.git' || name === 'dist' || name === 'build') continue;
    const full = path.join(dir, name);
    const stat = fs.statSync(full);
    if (stat.isDirectory()) {
      walk(full);
    } else if (name.endsWith('.go')) {
      const content = fs.readFileSync(full, 'utf8');
      if (content.includes(OLD)) {
        const count = (content.match(new RegExp(OLD.replace(/\//g, '\\/'), 'g')) || []).length;
        const newContent = content.split(OLD).join(NEW);
        fs.writeFileSync(full, newContent, 'utf8');
        changedFiles++;
        totalReplacements += count;
        console.log(`  fixed: ${path.relative(ROOT, full)} (${count}x)`);
      }
    }
  }
}

walk(ROOT);
console.log(`\nDone: ${changedFiles} files changed, ${totalReplacements} replacements total`);

File: scripts/fix_module_path.py
#!/usr/bin/env python3
"""批量替换所有 .go 文件中的旧 module 路径"""
import os

old = 'github.com/TeaOSLab/EdgeAdmin'
new = 'github.com/hujiali30001/freecdn-admin'
count = 0
files_changed = 0

root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
for root, dirs, files in os.walk(root_dir):
    dirs[:] = [d for d in dirs if d not in ('.git', 'dist', 'build')]
    for f in files:
        if not f.endswith('.go'):
            continue
        path = os.path.join(root, f)
        try:
            with open(path, 'r', encoding='utf-8', errors='ignore') as fh:
                content = fh.read()
            if old in content:
                new_content = content.replace(old, new)
                with open(path, 'w', encoding='utf-8', newline='') as fh:
                    fh.write(new_content)
                n = content.count(old)
                count += n
                files_changed += 1
                print(f'  fixed: {os.path.relpath(path, root_dir)} ({n} replacements)')
        except Exception as e:
            print(f'  error: {path}: {e}')

print(f'\nDone: {files_changed} files changed, {count} replacements total')
